packages/hardhat-core/src/internal/core/jsonrpc/types/base-types.ts

```diff
@@ -40,6 +41,14 @@ export const rpcStorageSlot = new t.Type<bigint>(
   t.identity
 );
 
+export const rpcStorageSlotHexStr = new t.Type<string>(
```

Comment:
Personal preference: `rpcStorageSlotHexString` (if you run `rg 'Str\b' src/` you'll see that we don't use this abbreviation often).